.travis.yml

```diff
@@ -37,7 +37,8 @@ matrix:
 before_install:
   - "export DISPLAY=:99.0"
   - "sh -e /etc/init.d/xvfb start"
-  - npm config set spin false
+  # if npm version is less than 3.0.0, upgrade to 3
+  - if [[ $(npm -v | cut -d '.' -f 1) -lt 3 ]]; then npm i -g npm@^3; fi
```

Comment:
we can do that too, the approach above is copied from what we do in most of the Ember CLI projects